pyclowder/api/v2/files.py

```diff
@@ -173,36 +171,69 @@ def upload_preview(connector, client, fileid, previewfile, previewmetadata=None,
     """
 
     connector.message_process({"type": "file", "id": fileid}, "Uploading file preview.")
-
     logger = logging.getLogger(__name__)
-    headers = {'Content-Type': 'application/json'}
 
-    # upload preview
-    url = '%s/api/previews?key=%s' % (client.host, client.key)
-    with open(previewfile, 'rb') as filebytes:
-        # If a custom preview file MIME type is provided, use it to generate the preview file object.
-        if preview_mimetype is not None:
-            result = connector.post(url, files={"File": (os.path.basename(previewfile), filebytes, preview_mimetype)},
-                                    verify=connector.ssl_verify if connector else True)
-        else:
-            result = connector.post(url, files={"File": filebytes}, verify=connector.ssl_verify if connector else True)
+    # upload visualization URL
+    visualization_url = '%s/api/v2/visualizations?name=%s&description=%s' % (
+        client.host, visualization_name, visualization_description)
```

Comment:
you might have to update that part to match the change in the endpoint

you'll have to swith the order you do things: 
- first register the config
- then post the visualization data with config id there

Reply:
Yep, I just did the above items. :-D